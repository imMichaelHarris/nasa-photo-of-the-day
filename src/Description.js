import React from 'react';

const Description = ({desc}) => {
    return (
        <div>
            <p>{desc}</p>
        </div>
    );
};

export default Description;