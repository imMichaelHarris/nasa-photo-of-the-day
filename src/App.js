import React, { useState, useEffect } from "react";
import axios from "axios";
import "./App.css";
import Media from "./Media";
import MediaInfo from "./MediaInfo";
import HDMedia from "./HDMedia";
import Description from "./Description";
function App() {
  const [media, setMedia] = useState([]);
  const [hd, setHD] = useState(false);

  const getAPOD = () => {
    axios
      .get(
        "https://api.nasa.gov/planetary/apod?api_key=fa5O3BQljsEyqfNarlgF5bX4t4TbK6vLbySh9i7H&date=2012-03-14"
      )
      .then(res => {
        console.log(res);
        setMedia(res.data);
      })
      .catch(err => console.log(err));
  };
  useEffect(() => getAPOD(), []);
  console.log(media);

  const showHD = () => setHD(!hd);
  

  return (
    <div className="App">
      
        <button onClick={showHD}>Show HD Photo</button>
        <MediaInfo title={media.title} date={media.date} desc={media.explanation}/>
        {hd ? <HDMedia media={media.hdurl}/> : <Media media={media.url} /> }
        <Description />
    </div>
  );
}

export default App;
