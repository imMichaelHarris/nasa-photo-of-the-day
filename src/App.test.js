import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer'
import App from './App';
import Description from './Description';


it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('<App />', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<App />).toJSON();

    expect(tree).toMatchSnapshot();
  });
  it('should match snapshot', () => {
    const tree = renderer.create(< Description/>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
