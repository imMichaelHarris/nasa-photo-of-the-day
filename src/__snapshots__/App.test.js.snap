// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match snapshot 1`] = `
<div
  className="App"
>
  <div
    className="ui middle aligned three column grid"
  >
    <div
      className="black stretched center aligned column"
    >
      <div>
        <h2 />
        <p />
      </div>
    </div>
    <div
      className="column"
    >
      <div>
        <img
          alt="Nasa Photo of the day"
          className="ui image"
        />
      </div>
    </div>
    <div
      className="black center aligned column"
    >
      <div>
        <p />
      </div>
    </div>
  </div>
  <div
    className="ui huge buttons"
  >
    <button
      className="ui button"
      onClick={[Function]}
    >
      HD
    </button>
    <div
      className="or"
    />
    <button
      className="ui button"
      onClick={[Function]}
    >
      SD
    </button>
  </div>
</div>
`;

exports[`<App /> should match snapshot 2`] = `
<div>
  <p />
</div>
`;
